### PR TITLE
feat: Return from value for replace operations

### PR DESCRIPTION
**Pull request by joshianiket22**
Updated code to return from value(old value) for replace operations

### CODE REVIEW

**Review by tjdavis3 — COMMENTED** [review]
@joshianiket22 should we use `from` or would it be better to use `oldValue` since from has a specific meaning for RFC6902?  Using oldValue would require:
- setting the struct tag to `json:"oldValue,omitempty"
- setting oldValue to "" in MarshalJSON for all Operations except `OperationReplace`

Thoughts?

**Comment by joshianiket22** [comment]
> @joshianiket22 should we use `from` or would it be better to use `oldValue` since from has a specific meaning for RFC6902? Using oldValue would require:
> 
>     * setting the struct tag to `json:"oldValue,omitempty"
> 
>     * setting oldValue to "" in MarshalJSON for all Operations except `OperationReplace`
> 
> 
> Thoughts?

I agree `from` has a specific meaning as per RFC6902 and it might be a possibility that the use case changes in future and we might also need to set `from` value in diff. For clarity purposes also OldValue makes more sense. Will make the required changes.

File: operation.go [hunk 1]
```diff
@@ -41,10 +40,12 @@ func (o Operation) MarshalJSON() ([]byte, error) {
 	switch o.Type {
 	case OperationCopy, OperationMove:
 		o.Value = nil
+		o.OldValue = nil
 	case OperationAdd, OperationTest:
 		o.From = emptyPtr
-	case OperationReplace:
-		o.From = pointer(fmt.Sprintf("%v", o.OldValue))
+		o.OldValue = nil
```
Comment (tjdavis3) [inline operation.go:46]: @joshianiket22 this shouldn't be here should it?  On a replace we WANT to show the oldValue.

Comment (joshianiket22) [reply operation.go:46]: I've already removed that. Its also not showing up in commit.

**Review by tjdavis3 — APPROVED** [review]

**Pull request merged**